org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/PersistentUpdater.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.workspace.internal;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.Collections;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+
+import org.eclipse.buildship.core.CorePlugin;
+import org.eclipse.buildship.core.preferences.PersistentModel;
+
+/**
+ * Base updater class to load and store updated item names.
+ *
+ * @author Donat Csikos
+ */
+public abstract class PersistentUpdater {
```

Comment:
Frankly, I like the superclass better because it states that from all the existing updaters only the subclasses persist their state.